### PR TITLE
Fixup #10308: Remove TOKmanifest from token.h

### PR DESCRIPTION
**Pull request by Geod24**
Original PR : #10308 
CC @kinke : Is the `TOK` change acceptable ? Otherwise I can also change `toChars` to take a `ubyte`. And thanks for reminding me of `tokens.h` existence.

### CODE REVIEW

File: src/dmd/target.d [hunk 1]
```diff
@@ -310,7 +310,7 @@ struct Target
      * Returns:
      *      true if the operation is supported or type is not a vector
      */
-    extern (C++) bool isVectorOpSupported(Type type, ubyte op, Type t2 = null)
+    extern (C++) bool isVectorOpSupported(Type type, TOK op, Type t2 = null)
```
Comment (kinke) [inline src/dmd/target.d]: The C++ declaration: https://github.com/dlang/dmd/blob/ac53452a31ad00dc9da1b6cab2e2c758644a6e1e/src/dmd/target.h#L91

So this won't work due to mangling differences between D `enum TOK : ubyte` and the new C++ `struct TOK`. I'd just leave it as it was.

File: src/tests/cxxfrontend.c [hunk 2]
```diff
@@ -92,6 +92,19 @@ static void frontend_term()
...
 /**********************************/
 
+void test_tokens()
+{
+    // First valid TOK value
+    assert(TOKlparen == 1);
+    assert(strcmp(Token::toChars(TOKlparen), "(") == 0);
+
+    // Last valid TOK value
+    assert(TOKvectorarray == 235);
```
Comment (kinke) [inline src/tests/cxxfrontend.c]: `== TOKMAX - 1`?

**Comment by Geod24** [comment]
@kinke : Fixed

**Review by thewilsonator — APPROVED** [review]

**Review by kinke — APPROVED** [review]

**Review by ibuclaw — APPROVED** [review]

File: src/tests/cxxfrontend.c [hunk 3]
```diff
@@ -92,6 +92,19 @@ static void frontend_term()
 
 /**********************************/
 
+void test_tokens()
+{
+    // First valid TOK value
+    assert(TOKlparen == 1);
```
Comment (ibuclaw) [inline src/tests/cxxfrontend.c]: This assert doesn't really test the C++<->D interface, but nevermind.

**Pull request merged**